chore: pin github action

lucasgonze opened the pull request:
Fixes https://github.com/magma/security/issues/97

Version numbers were converted to hashes using https://github.com/mheap/pin-github-action/. We'll need to periodically re-run this in the future, and might want a monthly action.

I had to modify `/magma/.github/workflows/config/yamllint_config.yml` to accept the YAML output formatting from that script, as follows:

```
15,20c15,16
<   braces: 
<     max-spaces-inside: 1
<     min-spaces-inside: 1
<   brackets:
<     min-spaces-inside: 0
<     max-spaces-inside: 1
---
>   braces: enable
>   brackets: enable
25d20
<     min-spaces-from-content: 1
31,32c26
<   empty-lines: 
<     max: 3
---
>   empty-lines: enable
43c37
<   trailing-spaces: disable
---
>   trailing-spaces: enable
```


<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary

<!-- Enumerate changes you made and why you made them -->

## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

LKreutzer commented on .github/workflows/agw-workflow.yml:
```diff
@@ -156,33 +156,36 @@ jobs:
           image: ${{ env.BAZEL_BASE_IMAGE }}
           # TODO: Remove work-around mount of Github workspace to /magma (https://github.com/addnab/docker-run-action/issues/11)
-          options: -v ${{ github.workspace }}:/workspaces/magma/ -v ${{ github.workspace }}/lte/gateway/configs:/etc/magma
+          options: -v ${{ github.workspace }}:/workspaces/magma/ -v ${{ github.workspace
```
This change looks somewhat strange - was this an auto-formatter? It looks like line 200 might even be longer and was not changed?

lucasgonze replied:
Right, this is the auto-formatter. 

I have changed the configuration to break lines at 200 characters.

LKreutzer commented on .github/workflows/agw-workflow.yml:
```diff
@@ -156,33 +156,36 @@ jobs:
-      - name: Run `bazel test //orc8r/gateway/c/...:* //lte/gateway/c/...:* --config=asan`
-        uses: addnab/docker-run-action@v3
+      - name: Run `bazel test //orc8r/gateway/c/...:* //lte/gateway/c/...:*
+          --config=asan`
```
This line break might be unnecessary - as above. Can we configure this?

LKreutzer commented on .github/workflows/agw-workflow.yml:
```diff
@@ -156,33 +156,36 @@ jobs:
           image: ${{ env.BAZEL_BASE_IMAGE }}
           # TODO: Remove work-around mount of Github workspace to /magma (https://github.com/addnab/docker-run-action/issues/11)
-          options: -v ${{ github.workspace }}:/workspaces/magma/ -v ${{ github.workspace }}/lte/gateway/configs:/etc/magma
+          options: -v ${{ github.workspace }}:/workspaces/magma/ -v ${{ github.workspace
```
As above.

lucasgonze replied:
Agreed. In the latest version lines are auto-wrapped at 200 characters.

LKreutzer commented on .github/workflows/amis-workflow.yml:
```diff
@@ -1,8 +1,8 @@
-on:  # yamllint disable-line rule:truthy
-  workflow_dispatch:
+on: # yamllint disable-line rule:truthy
+  workflow_dispatch: null
```
Why is this needed?

lucasgonze replied:
That is a default of the auto-formatter. See nullStr at https://eemeli.org/yaml/#tostring-options . I have changed the options to suppress that "null".

LKreutzer reviewed (approved):
The version pinning looks good!

Regarding the auto-formatting of the `name`, `args` and `options` fields: I find that in many cases the formatted version is less readable and it might be better to allow for a greater line length limit.

sebathomas commented on .github/workflows/cwf-integ-test.yml:
```diff
@@ -2,8 +2,9 @@
-on:  # yamllint disable-line rule:truthy
-  workflow_dispatch:
+on:
+  # yamllint disable-line rule:truthy
```
Did you move these yamllint comments accidentally (looks like it could be an autoformat issue)? It seems the lines now trigger the linter.

lucasgonze replied:
I replaced the case-by-case `truthy` configurations with one at the global level. We were either overriding it or ignoring the warning pretty much everywhere.

sebathomas replied:
I agree the check should be disabled globally, but it seems with the changes to these comments there are a lot of lint errors in this PR?

lucasgonze replied:
Could you help me find the lint warnings, @sebathomas ? I checked before submitting and found just the following, which I believe were present already:

```
$ yamllint *.{yml,yaml} -c config/yamllint_config.yml 
codeql-analysis.yml
  12:1      warning  missing document start "---"  (document-start)
  65:8      warning  missing starting space in comment  (comments)

gcc-problems.yml
  10:1      warning  missing document start "---"  (document-start)

golang-build-test.yml
  158:9     warning  comment not indented like content  (comments-indentation)
```

lucasgonze replied:
I'm seeing a reduction in linter errors. In the current master, `yamllint *.{yml,yaml} -c config/yamllint_config.yml ` produces 31 lines, including the "missing document start" above, and errors as well as warnings. In the new version I'm seeing 10 lines, all warnings, nothing new.

sebathomas replied:
So the reviewdog comments in the Changed Files are based on an old version of the PR and don't get removed automatically? That's what's confusing me.

Neudrino replied:
[The workflow is triggered `on: pull_request_target`](https://github.com/magma/magma/blob/master/.github/workflows/reviewdog-workflow.yml#L4). That means, that the actual file which is executed on this PR itself, is the code from current `master`, where the respective change is not yet active.

To remove the warnings here, one would need a two step process.
1. Changing the "yaml lint config".
2. Changing everything else.

lucasgonze replied:
I have created a single-purpose PR to change the yamllint config:

https://github.com/magma/magma/pull/13053

Neudrino replied:
> I have created a single-purpose PR to change the yamllint config:
> - #13053

#13053 was closed. This is ok, as the linter warnings are only in this PR, and we decided to ignore them here, as the configuration will be adopted.

Neudrino replied:
I should have made clearer that mentioning the 2 step process was a "theoretical workaround", which is not necessary to apply here now, imho.
My appologies, @lucasgonze !

Neudrino commented on .github/workflows/agw-docker-load-test.yaml:
```diff
@@ -1,7 +1,8 @@
 
-on:  # yamllint disable-line rule:truthy
+on:
+  # yamllint disable-line rule:truthy
```
Those changes should be reverted everywhere. It needs to be in the same line.

lucasgonze replied:
The disable-line statement is vestigial since adding `truthy: disable` to config/yamllint_config.yml. Whether it's on the same line has no impact. In the latest commit I have deleted all of the extra "truthy" lines.

lucasgonze replied:
This is moot because none of these lines has an impact since adding that same setting in the global configuration file. To make that more clear I have eliminated all the line-by-line "yamllint disable-line rule:truthy" statements.

Neudrino commented on .github/workflows/cwf-integ-test.yml:
```diff
@@ -2,8 +2,8 @@
-on:  # yamllint disable-line rule:truthy
-  workflow_dispatch:
+on:
+  workflow_dispatch: null
```
One of the previous comments (https://github.com/magma/magma/pull/12990#discussion_r899375308) sounds as if there should not be a `null` added here?

Neudrino commented on .github/workflows/config/yamllint_config.yml:
```diff
@@ -34,6 +39,5 @@ rules:
-  truthy:
-    level: warning
+  trailing-spaces: disable
+  truthy: disable
```
_For the record:_ This is the change, which should remove the "truthy warning". (See also https://github.com/magma/magma/pull/12990#discussion_r902820945)

Neudrino commented on .github/workflows/config/yamllint_config.yml:
```diff
@@ -34,6 +39,5 @@ rules:
-  trailing-spaces: enable
-  truthy:
-    level: warning
+  trailing-spaces: disable
```
Could you comment, why this change was necessary?
I would see it rather desired, to have no trailing spaces in the yaml files.

lucasgonze replied:
The yaml generator sometimes leaves trailing spaces. 

I have re-enabled the trailing spaces rule and edited all the generated code to prevent the warning. 

I have also cleaned up yaml formatting causing three other warnings, so we can have 100% clean yamllint output unless a new problem is introduced, which will make the output more useful:

1. comment not indented like content
2. missing starting space in comment
3. missing document start "---"

lucasgonze committed:
chore: pin github action

YAML formatting options: {'lineWidth':200,'nullStr':''}. In yamllint config, disable truthy warning, which was overriddin in almost 100% of the actual yaml, and delete corresponding line-by-line "truthy" statements in individual files. Consolidate versions of actions/upload-artifact, addnab/docker-run-action, actions/github-script. Fix all yamllint warnings.

Signed-off-by: Lucas Gonze <lucas@gonze.com>

tmdzk reviewed (approved):
Thanks for your service. this looks like a painful PR
